test/e2e/monitoring_stack_controller_test.go

```diff
@@ -182,6 +182,39 @@ func reconcileStack(t *testing.T) {
 	assert.DeepEqual(t, expected.ServiceMonitorSelector, generated.Spec.ServiceMonitorSelector)
 	assert.Equal(t, expected.LogLevel, generated.Spec.LogLevel)
 	assert.Equal(t, expected.Retention, generated.Spec.Retention)
+
```

Comment:
I think that we can leverage the status field more generally in the e2e tests: whenever we use `f.K8sClient.Create(...)` and expect a success, we should rewrite it to `f.K8sClient.CreateAndWaitAvailable(...)` which would create the stack and wait for Available/Reconciled to be True. I'm not sold on the name of the function though ;)

Reply:
Yes something like `GetAndAwait`. Perhaps I can create some function in the test framework or assertions.